src/arrayMethodJoin.js

```diff
@@ -4,8 +4,42 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let joinedString = '';
+
+    if (this.length === 0) {
+      return '';
+    }
+
+    if (this.length === 1) {
+      return '' + this[0];
+    }
+
+    if (this[1] === undefined) {
+      return this;
+    }
+
+    for (let i = 0; i < this.length - 1; i++) {
+      if (this[i] === undefined || this[i] === null) {
+        joinedString += '';
+      } else {
+        joinedString += this[i];
+      }
+
+      if (separator === null) {
+        joinedString += 'null';
+      }
+
+      if (separator !== null) {
+        joinedString += separator;
+      }
+    }
```

Comment:
you can use template string instead of checking for null and assigning the string `null` to joinedString. Template string will convert null to 'null' on its own